pkg/platform/kube/platform.go

```diff
@@ -801,13 +821,36 @@ func (p *Platform) CreateAPIGateway(ctx context.Context,
 
 // UpdateAPIGateway will update a previously existing api gateway
 func (p *Platform) UpdateAPIGateway(ctx context.Context, updateAPIGatewayOptions *platform.UpdateAPIGatewayOptions) error {
+	// get existing api gateway
 	apiGateway, err := p.consumer.NuclioClientSet.NuclioV1beta1().
 		NuclioAPIGateways(updateAPIGatewayOptions.APIGatewayConfig.Meta.Namespace).
 		Get(ctx, updateAPIGatewayOptions.APIGatewayConfig.Meta.Name, metav1.GetOptions{})
 	if err != nil {
 		return errors.Wrap(err, "Failed to get api gateway to update")
 	}
 
+	// restore existing config
+	apiGatewayConfig := &platform.APIGatewayConfig{
```

Comment:
Use `getAPIGatewayConfigFromCRD`

Reply:
@TomerShor it's impossible to avoid cycled imports here

Reply:
Then move `getAPIGatewayConfigFromCRD` to `platform/types.go` below `GetAPIGatewayConfigFromInterface`.

While you're at it - change `functionConfigInterface` to apiGatewayConfigInterface` in [here](https://github.com/nuclio/nuclio/blob/development/pkg/platform/types.go#L484-L485)